contracts/VestingSwap.sol

```diff
@@ -57,27 +57,18 @@ contract VestingSwap is Ownable, ReentrancyGuard {
     }
 
     function startAlphaSwap() external onlyOwner nonReentrant {
-        swap[alphaPresale].sold = IPresale(alphaPresale).totalSold();
-        require(IERC20(HBT).balanceOf(address(this)) == swap[alphaPresale].sold, "VestingSwap: HBT_NOT_ALLOCATED_FOR_ALPHA");
+        uint requiredBalance = ALPHA_PRESALE_LIMIT.add(BETA_PRESALE_LIMIT).add(GAMMA_PRESALE_LIMIT); 
```

Comment:
Do you mint all HBT for swap at once?
Why you do not check if balance is sufficient for the presales other than Alpha?
Do you check anywhere if the swap amount of sHBT can exceed the corresponding amount of HBT for particular presale?

Reply:
HBT allocated once, because we know total amount for swaps. All swaps start sequentially (this is checked by modifiers), so only one check in for allocation is needed (for Alpha start). All checks regarding sHBT limits live in `_swap` function